commit 4daf2d02ed2532572872e8cf6eb4915e5a8af53a
Author: Alexander Awitin <alexanderawitin@gmail.com>
Date:   Wed Feb 20 16:06:02 2019 +0800

    [stable/redis] Fix "iptables: Permission Denined" error when installing on Istio-enabled cluster. (#11232)
    
    * [stable/redis] Fix "Can't initialize iptables table 'nat': Permission denied (you must be root)" error when installed on an Istio-enabled cluster.
    
    Only define the `securityContext` on the main container instead of defining it on the top level `spec` which results into injected containers by Istio inheriting this definition (i.e. istio-init).
    
    Related topic: https://github.com/istio/old_issues_repo/issues/316
    
    Signed-off-by: Alexander Awitin <alexanderawitin@gmail.com>
    
    * [stable/postgresql] Fix "Can't initialize iptables table 'nat': Permission denied (you must be root)" error when installed on an Istio-enabled cluster. (#11226)
    
    Only define the `securityContext` on the main container instead of defining it on the top level `spec` which results into injected containers by Istio inheriting this definition (i.e. istio-init).
    
    Related topic: https://github.com/istio/old_issues_repo/issues/316
    
    Signed-off-by: Alexander Awitin <alexanderawitin@gmail.com>
    
    * Bring `securityContext.fsGroup` back to `spec.template.spec` as where it should be.
    Ref: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#podsecuritycontext-v1-core
    
    Signed-off-by: Alexander Awitin <alexanderawitin@gmail.com>
    
    * refactor: Split `runAsUser` and `fsGroup` configuration in container and pod sections respectively.
    
    Signed-off-by: Alexander Awitin <alexanderawitin@gmail.com>
    
    * Bump chart version.
    
    Signed-off-by: Alexander Awitin <alexanderawitin@gmail.com>
    
    * refactor: Remove `securityContext.runAsNonRoot` from the main container.
    
    Signed-off-by: Alexander Awitin <me@alexanderawitin.com>

diff --git a/stable/redis/Chart.yaml b/stable/redis/Chart.yaml
index dff5aaddc..b6d925c29 100644
--- a/stable/redis/Chart.yaml
+++ b/stable/redis/Chart.yaml
@@ -1,5 +1,5 @@
 name: redis
-version: 6.1.0
+version: 6.1.1
 appVersion: 4.0.12
 description: Open source, advanced key-value store. It is often referred to as a data structure server since keys can contain strings, hashes, lists, sets and sorted sets.
 keywords:
diff --git a/stable/redis/templates/redis-master-statefulset.yaml b/stable/redis/templates/redis-master-statefulset.yaml
index f0558db60..1f87c0dba 100644
--- a/stable/redis/templates/redis-master-statefulset.yaml
+++ b/stable/redis/templates/redis-master-statefulset.yaml
@@ -41,7 +41,6 @@ spec:
       {{- if .Values.master.securityContext.enabled }}
       securityContext:
         fsGroup: {{ .Values.master.securityContext.fsGroup }}
-        runAsUser: {{ .Values.master.securityContext.runAsUser }}
       {{- end }}
       serviceAccountName: "{{ template "redis.serviceAccountName" . }}"
       {{- if .Values.master.priorityClassName }}
@@ -66,6 +65,10 @@ spec:
       - name: {{ template "redis.fullname" . }}
         image: "{{ template "redis.image" . }}"
         imagePullPolicy: {{ default "" .Values.image.pullPolicy | quote }}
+        {{- if .Values.master.securityContext.enabled }}
+        securityContext:
+          runAsUser: {{ .Values.master.securityContext.runAsUser }}
+        {{- end }}
         command:
         - /bin/bash
         - -c
diff --git a/stable/redis/templates/redis-slave-deployment.yaml b/stable/redis/templates/redis-slave-deployment.yaml
index 981290a10..54bc9d948 100644
--- a/stable/redis/templates/redis-slave-deployment.yaml
+++ b/stable/redis/templates/redis-slave-deployment.yaml
@@ -45,8 +45,15 @@ spec:
         - name: {{ . }}
         {{- end}}
       {{- end}}
-      {{- /* Include master securityContext if slave securityContext not defined */ -}}
-      {{ include "redis.slave.securityContext" . | indent 6 }}
+      {{- /* Include master securityContext.fsGroup if slave securityContext.fsGroup not defined */ -}}
+      {{- if (.Values.slave.securityContext | default .Values.master.securityContext) }}
+      securityContext:
+        {{- if .Values.slave.securityContext }}
+        fsGroup: {{ .Values.slave.securityContext.fsGroup }}
+        {{- else }}
+        fsGroup: {{ .Values.master.securityContext.fsGroup }}
+        {{- end }}
+      {{- end }}
       serviceAccountName: "{{ template "redis.serviceAccountName" . }}"
       {{- if .Values.slave.priorityClassName }}
       priorityClassName: "{{ .Values.slave.priorityClassName }}"
@@ -71,6 +78,15 @@ spec:
         image: {{ template "redis.image" . }}
         imagePullPolicy: {{ .Values.image.pullPolicy | default "" | quote }}
 {{- $command := default .Values.master.command .Values.slave.command }}
+        {{- /* Include master securityContext.runAsUser if slave securityContext.runAsUser not defined */ -}}
+        {{- if (.Values.slave.securityContext | default .Values.master.securityContext) }}
+        securityContext:
+          {{- if .Values.slave.securityContext }}
+          runAsUser: {{ .Values.slave.securityContext.runAsUser }}
+          {{- else }}
+          runAsUser: {{ .Values.master.securityContext.runAsUser }}
+          {{- end }}
+        {{- end }}
         command:
         - /bin/bash
         - -c
